### PR TITLE
kind: Bump k8s version to 1.27.0

### PR DESCRIPTION
**Pull request by sayboras**

### CODE REVIEW

**Comment by tklauser** [comment]
Should we incorporate these updates into https://docs.cilium.io/en/latest/contributing/development/dev_setup/#add-update-a-new-kubernetes-version and update kind k8s version along with all the other tests?

**Comment by sayboras** [comment]
> Should we incorporate these updates into https://docs.cilium.io/en/latest/contributing/development/dev_setup/#add-update-a-new-kubernetes-version and update kind k8s version along with all the other tests?

Good point, I didn't check this page before.

**Review by bimmlerd — APPROVED** [review]

**Comment by sayboras** [comment]
This PR only changes kind config files, which are only used in GHA tests, so full CI is not required.

File: Documentation/contributing/development/dev_setup.rst [hunk 1]
```diff
@@ -589,6 +590,8 @@ Minor version
 #. Add the new version in ``test/provision/k8s_install.sh``, if it is an RC
    install it using binaries.
 
+#. Bump the kindest/node version in all of kind's config files ``.github/kind-config*``.
```
Comment (zacharysarah) [inline Documentation/contributing/development/dev_setup.rst]: Nit
```suggestion
#. Bump the kindest/node version in all of kind's config files (for example, ``.github/kind-config*``).
```

**Review by zacharysarah — APPROVED** [review]
@sayboras Looks good from a docs perspective. ✨ One nit, otherwise LGTM. I'm approving preemptively with the understanding that you'll fix prior to merge. 🙇🏻

**Comment by sayboras** [comment]
The changes are only for kind configuration, which is then used in GHA jobs only, so full CI is not required. 

Reviews are in, marking this ready to merge.

**Pull request merged**